andela/ninjas-ah-backend
# #165412860 add chat feature

#### What does this PR do?
Add the implementation of a basic chat with a permanent data storage
#### Description of Task to be completed?
Have the following endpoints accessible to an authenticated user:
- `POST /api/v1/chats` : send a message
- ` GET /api/v1/chats` : get all chats. To filter chats the following parameters can be appended on the URL: `?limit=<number>` or `?offset=<number`
- `DELETE /api/v1/chats/:chatId` : to delete a specific chat
#### How should this be manually tested?
- After cloning the repo, cd into it and type `npm test`
- Using Postman, firstly make sure you signup then login to get the access token, secondly make sure you pass the access token in the header with the following key name: `access-token`, finally send a JSON object of the following structure : `{"message":"'your message"}`
- Using the mockup,  after running the server by typing `npm run start` or `npm run dev`, firstly create an account by visiting `http://localhost:<PORT>/mockups/signup.html`, then login by visiting `http://localhost:<PORT>/mockups/login.html` to get a token, finally visit `http://localhost:<PORT>/mockups/chat.html` to start chatting
#### What are the relevant pivotal tracker stories?
[165412860](https://www.pivotaltracker.com/story/show/165412860)
#### Screenshots
![image](https://user-images.githubusercontent.com/44000547/58168493-b37c9f80-7c8e-11e9-8a1e-4ead930dedb8.png)
